src/components/timeline/index.tsx

```diff
@@ -0,0 +1,288 @@
+import React, {useCallback, useMemo, useEffect, useState} from 'react';
+import {StyleSheet, ImageRequireSource, ViewStyle, MeasureOnSuccessCallback, LayoutChangeEvent} from 'react-native';
+import {Colors, Spacings} from '../../style';
+import View from '../view';
+import Icon from '../icon';
+import Text from '../text';
+import Dash from './Dash';
+
+
+const LINE_WIDTH = 2;
+const POINT_SIZE = 12;
+const HALO_WIDTH = 4;
+const HALO_TINT = 70;
+const HOLLO_WIDTH = 2;
+const CONTENT_POINT_SIZE = 20;
+const ICON_SIZE = 12;
+const CONTENT_CONTAINER_PADDINGS = Spacings.s2;
+const POINT_MARGINS = Spacings.s1;
+
+export enum StateTypes {
+  CURRENT = 'current', // default
+  NEXT = 'next',
+  ERROR = 'error',
+  SUCCESS = 'success'
+}
+
+export enum LineTypes {
+  FULL = 'full', // default
+  DASHED = 'dashed'
+}
+
+export enum PointTypes {
+  BULLET = 'bullet', // default
+  HOLLOW = 'hollow',
+  HALO = 'halo'
+}
+
+export type LineProps = {
+  state?: StateTypes;
+  type?: LineTypes;
+  color?: string;
+  /** to mark as entry point */
+  entry?: boolean;
+}
+
+export type PointProps = {
+  state?: StateTypes;
+  type?: PointTypes;
+  color?: string;
+  icon?: ImageRequireSource;
+  label?: number;
+  /** to align point to this view's center */
+  alignmentTargetRef?: React.MutableRefObject<undefined>;
+  /** the target view's top parent view */
+  targetContainerRef?: React.MutableRefObject<undefined>;
+}
+
+type Position = {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+};
+
+export type TimelineProps = {
+  height: number;
+  topLine?: LineProps;
+  bottomLine?: LineProps;
+  point?: PointProps;
+  renderContent?: (props: TimelineProps) => JSX.Element;
```

Comment:
I see your point. I'll change the design, maybe it will help with the height and containerRef as well

Reply:
I'm afraid neither the height nor the ref issues were resolved this way... Looking for other solutions...

Reply:
Fixed

Reply:
Great. 
Notice that you need to wrap the `TimelineProps` with React's `PropsWithChildren` util 
Your build fails, probably because of TS errors related to that

Reply:
Done